### PR TITLE
Fire an input event on `editor.action.clipboardPasteAction`

### LINKED ISSUE #186998: Right click to paste in input dialogs does not trigger an update
```
Version: 1.79.2
Commit: 695af097c7bd098fbf017ce3ac85e09bbc5dda06
Date: 2023-06-14T08:55:41.903Z (2 wks ago)
Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
```

Steps to Reproduce:

1. Add `"workbench.quickOpen.closeOnFocusLost": false` to settings.json so you can right click in dialogs without them loosing focus (this is a separate issue).
1. Copy some content that you will want to search, eg a file path.
1. Open a dialog, tested in Quick Open and the Command Pallete.
1. Right click in the dialog and paste the contents.

The expectation is that this would paste and update the dialog results to the contents you have pasted. This *is* what happens if you paste with `control + v`.

However, it pastes the contents, but the results are not updated until you edit the input in some other way.

image

### PR DESCRIPTION
**Pull request by a-stewart**
This fixes #186998

In VS Code web, when `editor.action.clipboardPasteAction` is triggered (eg by right clicking on an input and selecting paste) it updates the value by setting `element.value = ...`. However, this does not trigger the oninput handler which a the QuickInputs use to detect changes. As such right click and paste does not update the input causing #186998.

This can be fixed by updating the paste action to dispatch an input event on the element.

### CODE REVIEW

**Comment by a-stewart** [comment]
^ The issue is that Event is imported so the constructor clashes (but somehow still works?)

I'll update the code to avoid the collision.

**Comment by a-stewart** [comment]
Updated import and tests now pass

File: src/vs/workbench/browser/actions/textInputActions.ts [hunk 1]
```diff
@@ -64,6 +64,7 @@ export class TextInputActionsProvider extends Disposable implements IWorkbenchCo
 						element.value = `${element.value.substring(0, selectionStart)}${clipboardText}${element.value.substring(selectionEnd, element.value.length)}`;
 						element.selectionStart = selectionStart + clipboardText.length;
 						element.selectionEnd = element.selectionStart;
+						element.dispatchEvent(new Event('input'));
```
[the remark this answers is not in the document]
Comment (a-stewart) [reply src/vs/workbench/browser/actions/textInputActions.ts]: Updated to add bubbled and cancelable here too.

[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply src/vs/workbench/browser/actions/textInputActions.ts]: Thanks @a-stewart. Can you also try another browser or two? Ideally:
* Safari
* Firefox
* Edge or Chrome

[the remark this answers is not in the document]
Comment (a-stewart) [reply src/vs/workbench/browser/actions/textInputActions.ts]: Hi,

Apologies for the delay on this.

We have been running this change on our internal fork for the past 2 weeks with users of multiple OSs and browsers and have had no reports of any issues.

I can also confirm that it works as expected in Chrome and Firefox (although granting clipboard permissions in Firefox seems very involved)

**Review by TylerLeonhardt — APPROVED** [review]

File: src/vs/workbench/browser/actions/textInputActions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply src/vs/workbench/browser/actions/textInputActions.ts]: Thanks for contributing!

**Review by lramos15 — APPROVED** [review]

**Pull request merged**